drivers/clock_control/clock_control_smartbond.c

```diff
@@ -145,10 +146,59 @@ static void smartbond_start_xtal32k(void)
 	}
 }
 
+#ifdef CONFIG_REGULATOR
+/*
+ * Should be used to control PLL when the regulator driver is available.
+ * If the latter is available, then the VDD level should be changed when
+ * switching to/from PLL. Otherwise, the VDD level is considered to
+ * be fixed @1.2V which should support both XTAL32M and PLL system clocks.
+ */
+static int smartbond_clock_set_pll_status(bool status)
+{
+	const struct device *dev  = DEVICE_DT_GET(DT_NODELABEL(vdd));
+	int ret;
+
+	if (!device_is_ready(dev)) {
+		LOG_ERR("Regulator device is not ready");
+		return -ENODEV;
+	}
+
+	if (status) {
+		/* Enabling PLL requires that VDD be raised to 1.2V */
+		if (regulator_set_voltage(dev, 1200000, 1200000) == 0) {
```

Comment:
To complete the answer, it would be burning energy without practical value for the user. This is designed in the same way in Renesas SDK and so far not user has requested manual control.

Reply:
Thanks.
In the long run will you be keeping with with / without regulator driver option, or will you be deprecating the non regulator method?

Reply:
In a regular design, regulators should be the default option along with PM subsystem to achieve optimum power consumption, which is critical for low-power applications. However, there might be applications that do not employ the PM subsystem and the default power supply scheme is enough (no regulators are required). I do not find any benefit in constraining users as long as the SoC can operate without issues in both cases.